### DISCUSSION TITLE
New histogram: string formats

### DISCUSSION THREAD

**Discussion by dlemstra** [ImageMagick/ImageMagick#5375] [Announcements]
As of IM `7.1.0.44`, there are 3 new string format that return the clip values from `-contrast-stretch`, `-normalize` and `-linear-stretch`. The returned black-pt and white-pt graylevel percents subsequently can be used by `-level` to stretch the histogram of any image. If `-level` is applied to the same images, then the results will be identical to running the method.

Note, however, that those values may not produce the same results on other images as running these 3 methods directly, because the black-pt and white-pt graylevel percents depend upon the shape of the histogram tails of the actual images.

Note also that `-normalize` uses the `%[histogram:contrast-stretch]` string format, since `-normalize` is actually just a short-hand for `-contrast-stretch 2x1%`

For example:

```
magick rose: -contrast-stretch 0x0% -format "%[histogram:contrast-stretch]\n" info:
14.4778x100%

magick rose: -normalize -format "%[histogram:contrast-stretch]\n" info:
17.5372x99.9145%

magick rose: -linear-stretch 2x2% -format "%[histogram:linear-stretch]\n" info:
17.5372x99.5514%

```

**Comment by snibgo** [comment]
Excellent, many thanks for this.
